### PR TITLE
Feature/client/A3 House Setting (Create)

### PR DESCRIPTION
**Pull request by kanatagu**
## Describe your changes
* Add house setting create page and functionality

To show create house page, you need to register a new account first. 

## Issue
https://github.com/HAK2024/fairy-share/issues/20

## URL
http://localhost:3000/house/create

## Screenshot (if you made the UI)
screencapture-localhost-3000-house-create-2024-03-19-15_53_43

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings and no errors

### CODE REVIEW

File: client/app/_components/ui/Form.tsx [hunk 1]
```diff
@@ -85,17 +85,22 @@ FormItem.displayName = 'FormItem'
 
 const FormLabel = React.forwardRef<
   React.ElementRef<typeof LabelPrimitive.Root>,
-  React.ComponentPropsWithoutRef<typeof LabelPrimitive.Root>
->(({ className, ...props }, ref) => {
+  React.ComponentPropsWithoutRef<typeof LabelPrimitive.Root> & {
+    isRequired?: boolean
```
Comment (kanatagu) [inline client/app/_components/ui/Form.tsx:89]: I added `isRequired` props to show required mark. Please add it to the components where you made whenever you can!

File: client/app/house/components/DynamicRuleField.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    control={control}
+                    name={`rules.${index}.text`}
+                    render={({ field }) => (
+                      <FormItem className='w-full'>
+                        <FormControl>
+                          <Input
+                            placeholder='Clean the kitchen after you cook'
+                            {...field}
+                            value={value[index]?.text || ''}
+                            className=''
```
Comment (Atsuki02) [inline client/app/house/components/DynamicRuleField.tsx:50]: Empty className!

File: client/app/house/hooks/useCreateHouse.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const router = useRouter()
+  const { toast } = useToast()
+
+  const { mutate, isPending } = useCreateHouseMutation()
+
+  const form = useForm<CreateHouseSchema>({
+    resolver: createHouseResolver,
+    defaultValues: {
+      name: '',
+      isExpensePerTime: undefined,
```
Comment (Atsuki02) [inline client/app/house/hooks/useCreateHouse.ts:20]: I'm also going to have to fix this part in my task form, but the document says that undefined should not be defaultValues.
https://www.react-hook-form.com/api/useform/#defaultValues

**Review by Atsuki02 — APPROVED** [review]
LGTM! I made some comments!

**Review by hyejin-seok — APPROVED** [review]

File: client/app/house/components/DynamicRuleField.tsx (hunk 2 above)
Comment (kanatagu) [reply client/app/house/components/DynamicRuleField.tsx]: Deleted it!
https://github.com/HAK2024/fairy-share/pull/81/commits/739296a060edd606f24d5b8c1e068dfde595d94b

File: client/app/house/hooks/useCreateHouse.ts (hunk 3 above)
Comment (kanatagu) [reply client/app/house/hooks/useCreateHouse.ts:20]: I'll fix it when I implement the edit form!

**Pull request merged**